.claude/specs/message-queue-system/PROPOSAL_INTEGRATOR.md

```
@@ -0,0 +1,983 @@
+POST   /threads/{thread_id}/abort   # Abort running task
+GET    /threads/{thread_id}/interrupts  # HITL interrupts
+POST   /threads/{thread_id}/resume      # Resume from interrupt
+```
+
+**Frontend Service Pattern (from `/frontend/src/lib/services/threadService.ts`):**
+- Functions exported individually: `searchThreads`, `deleteThread`, `abortThread`
+- Uses `apiClient` (axios instance) for HTTP calls
+- Returns typed responses aligned with backend Pydantic schemas
+- Handles authentication via `getAuthToken()` in headers
+
+### 2.2 New Endpoints Required for Queue Management
```

markdownlint-cli2 (0.18.1)

Unordered list indentation
Expected: 0; Actual: 2

(MD007, ul-indent)